projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -1599,7 +1605,11 @@ protected function get_related_post_ids( $post_id, $size, array $filters ) {
 			}
 		}
 
-		$results       = json_decode( wp_remote_retrieve_body( $response ), true );
+		$results = json_decode( wp_remote_retrieve_body( $response ), true );
+		if ( isset( $results['error'] ) && 'Conflict' === $results['error'] ) {
+			// Don't cache conflict errors, even for 60 seconds.
+			return array();
+		}
```

Comment:
Will we get to this point? I would think that the new return you've added in D86909-code would be caught just above, in [the `is_wp_error` check](https://github.com/Automattic/jetpack/pull/25946/files#diff-9105b068cf482dd9007ac0a2ff0985bf13dac7d398fb802678224448b5897841R1600), but I may be missing something.

Reply:
@jeherve Thanks for reviewing this!

From what I can see the `is_wp_error` check only catches errors like connection problem, timoeut etc. It doesn't handle cases where the response is retrieved successfully, but the response code is outside the 2xx/3xx range.

I was looking at how other packages handle that, and I found that there's also a function `wp_remote_retrieve_response_code` that can make this a bit more readable. Now, we will also return results from stale cache if available, instead of empty. What do you think about the updated condition?

I've also made a screenshot while xdebugging this to show that when responding with this new errror, the `$response` is not a `WP_Error`, it's an array:

![image](https://user-images.githubusercontent.com/6437642/187694136-a71125fe-459b-4991-8589-856ee77ef571.png)

Reply:
👍  Thanks for the extra details!